### ISSUE TITLE
`vim.lsp.util.get_progress_messages` will be removed in neovim 0.11

### ISSUE THREAD

**Issue by jinzhongjia** [beauwilliams/statusline.lua#20]
According this: [here](https://neovim.io/doc/user/deprecated.html#_lsp-functions)

`vim.lsp.util.get_progress_messages()` Use [vim.lsp.status()](https://neovim.io/doc/user/lsp.html#vim.lsp.status()) or access progress of [vim.lsp.client](https://neovim.io/doc/user/lsp.html#vim.lsp.client)

maybe I will open a pr to fix this

**Comment by jinzhongjia** [comment]
code is located at https://github.com/beauwilliams/statusline.lua/blob/20ad26912935f91918da9f428761b6d1b651d632/lua/sections/_lsp.lua#L61C1-L67

**Comment by beauwilliams** [comment]
Thanks for letting me know. I will see about getting it patched up. If you want to write a PR feel free!

**Comment by mariogarridopt** [comment]
I think this issue should be open so people know that this is still something that needs work.

As of now I still get the message every time I open nvim.

**Comment by mariogarridopt** [comment]
We can't just switch from `vim.lsp.util.get_progress_messages()` to `vim.lsp.status()` because the return changed from `array` to `string`.

This new string return does not carry any progress data.

Moving forward it should be possible to redraw the statusline on `LspProgress` events:

> `LspProgress`
> Upon receipt of a progress notification from the server. Notifications can
> be polled from a `progress` ring buffer of a `|vim.lsp.client|` or use
> `|vim.lsp.status()|` to get an aggregate message.
> -- https://github.com/neovim/neovim/blob/4e34ca8a/runtime/doc/lsp.txt#L593-L607

Example of message returned by various LSP's:

```viml
:lua for msg in vim.lsp.get_active_clients()[1].progress do vim.print(msg) end
```

**Comment by mariogarridopt** [comment]
It looks like the maintainer of this project is no longer active. If you are experiencing the same issue, you can use my fork of the project @ #21

**Comment by beauwilliams** [comment]
Hey @mariogarridopt I'm happy to merge this in, sorry for being MIA!

I've been so busy I'm still on 0.9 and haven't had the time to keep up lately. 

I see there is another pr here too - https://github.com/beauwilliams/statusline.lua/pull/22

I'll merge yours in as it looks good to me and we can close out this issue 🙂

**Comment by beauwilliams** [comment]
I have just merged in #21 thanks to @mariogarridopt
